File: test_api.php
<?php

// Simple API test script
$baseUrl = 'http://localhost:8000/api';

// Test login
echo "Testing Login...\n";
$loginData = [
    'email' => 'admin@carwash.com',
    'password' => 'password'
];

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $baseUrl . '/login');
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($loginData));
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Content-Type: application/json',
    'Accept: application/json'
]);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

$response = curl_exec($ch);
$httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

echo "Login Response (HTTP $httpCode):\n";
echo $response . "\n\n";

if ($httpCode === 200) {
    $loginResponse = json_decode($response, true);
    $token = $loginResponse['data']['token'];
    
    // Test dashboard
    echo "Testing Dashboard...\n";
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $baseUrl . '/dashboard');
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'Authorization: Bearer ' . $token,
        'Accept: application/json'
    ]);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    
    $dashboardResponse = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    
    echo "Dashboard Response (HTTP $httpCode):\n";
    echo $dashboardResponse . "\n\n";
    
    // Test services
    echo "Testing Services...\n";
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $baseUrl . '/services');
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'Authorization: Bearer ' . $token,
        'Accept: application/json'
    ]);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    
    $servicesResponse = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    
    echo "Services Response (HTTP $httpCode):\n";
    echo $servicesResponse . "\n\n";
}
?>

File: carwashapk/washmanager_mobile/lib/test_api.dart
import 'package:dio/dio.dart';

void main() async {
  final dio = Dio(BaseOptions(
    baseUrl: 'http://127.0.0.1:8000/api',
    connectTimeout: const Duration(seconds: 10),
    receiveTimeout: const Duration(seconds: 10),
    headers: {
      'Content-Type': 'application/json',
      'Accept': 'application/json',
    },
  ));

  try {
    print('🔄 Testing API connection...');
    
    // Test login
    final response = await dio.post('/login', data: {
      'email': 'admin@carwash.com',
      'password': 'password',
    });
    
    print('✅ API Response Status: ${response.statusCode}');
    print('📦 API Response Data: ${response.data}');
    
  } catch (e) {
    print('❌ API Test Failed: $e');
    if (e is DioException) {
      print('❌ DioException Type: ${e.type}');
      print('❌ Response: ${e.response?.data}');
    }
  }
}